code/modules/projectiles/guns/energy/nuclear.dm

```diff
@@ -52,27 +52,32 @@
 /obj/item/weapon/gun/energy/gun/nuclear/get_cell()
 	return DEVICE_NO_CELL
 
-/obj/item/weapon/gun/energy/gun/nuclear/proc/failcheck()
-	lightfail = 0
-	if (prob(src.reliability)) return 1 //No failure
-	if (prob(src.reliability))
-		for (var/mob/living/M in range(0,src)) //Only a minor failure, enjoy your radiation if you're in the same tile or carrying it
-			if (src in M.contents)
-				M << "<span class='warning'>Your gun feels pleasantly warm for a moment.</span>"
-			else
-				M << "<span class='warning'>You feel a warm sensation.</span>"
-			M.apply_effect(rand(3,120), IRRADIATE)
-		lightfail = 1
+/obj/item/weapon/gun/energy/gun/nuclear/small_fail()
+	for (var/mob/living/M in range(0,src)) //Only a minor failure, enjoy your radiation if you're in the same tile or carrying it
+		if (src in M.contents)

```

Comment:
No. See my comment above about passing `user` down to these procs. This would allow you to skip this very costly `if (src in M.contents)` check and simply allow you to do `user << "Your gun assploded."`